travis/before_install.sh

```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+
+# Don't source setup.sh here, because the virtualenv might not be set up yet
+
+# Install dependencies
+# removing the old c++ link is needed because otherwise brew install fails
+rm '/usr/local/include/c++'
+brew install ccache protobuf gcc@5
+
+# Install Python.
+# The brew update/upgrade commands are needed because otherwise homebrew
+# exits with an error code when installing python2 since it is already installed
+# but with a different version.
+if [ "${PYTHON_VERSION}" != "python2" ]; then
+  brew install ${PYTHON_VERSION}
+fi
+
+pip install virtualenv
```

Comment:
nit: By using the brew python, you don't need a virtualenv.

Reply:
I'll still keep it because it is more similar to an actual dev environment. Also, it makes the script simpler when I'm adding linux CI support (coming in a future PR)